src/web/BL_Python/web/config.py

```diff
@@ -8,6 +8,7 @@
 
 class LoggingConfig(BaseModel):
     log_level: str = "INFO"
+    format: Literal["plaintext", "JSON"] = "JSON"
```

Comment:
Is this safe? This constraint won't do anything if the value is loaded from some external source (a config or .env file, for instance). Sorry if you've already thought of this.

Reply:
I'm glad you asked! This exact concern is why I'm using Pydantic for the various Config classes. Here's how Pydantic handles [Literal](https://docs.pydantic.dev/1.10/usage/types/#literal-type).

```python
from typing import Literal
from pydantic import BaseModel

class FooModel(BaseModel):
    prop: Literal["FOO", "BAR"] = "FOO"

_ = FooModel(prop="FOO")
_ = FooModel(prop="BAR")
try:
    _ = FooModel(prop="BAZ")
except Exception as e:
    print(e)
```

```sh
1 validation error for FooModel
prop
  unexpected value; permitted: 'FOO', 'BAR' (type=value_error.const; given=BAZ; permitted=('FOO', 'BAR'))
```